evap/staff/views.py

```diff
@@ -626,40 +652,65 @@ def course_participant_import(request, semester_id, course_id):
     course = get_object_or_404(Course, id=course_id, semester=semester)
     raise_permission_denied_if_archived(course)
 
-    form = UserImportForm(request.POST or None, request.FILES or None)
+    excel_form = UserImportForm(request.POST or None, request.FILES or None)
+    copy_form = CourseParticipantCopyForm(request.POST or None, request.FILES or None)
 
-    if form.is_valid():
+    if request.method == "POST":
         operation = request.POST.get('operation')
-        if operation not in ('test', 'import'):
-            raise SuspiciousOperation("Invalid POST operation")
 
-        # Extract data from form.
-        excel_file = form.cleaned_data['excel_file']
-        import_course = form.cleaned_data['course']
+        if operation not in ('test', 'import', 'copy'):
+            raise SuspiciousOperation("Invalid POST operation")
 
         test_run = operation == 'test'
+        import_run = operation == 'import'
+        copy_run = operation == 'copy'
+
+        if test_run and excel_form.is_valid():
+            excel_file = excel_form.cleaned_data['excel_file']
+            imported_users, warnings, errors = UserImporter.process(request, excel_file, test_run)
```

Comment:
Changing the API would result in the need of always creating a WhateverImporter object so in the end, when rendering, you can use WhateverImporter.warnings etc., even when there was no test or import request. I don't know whether this would be beneficial. Fixed the underscore issue for now.